documentation/src/main/asciidoc/reference/search-dsl-highlighting.asciidoc

```diff
@@ -0,0 +1,416 @@
+== Highlighter type
+
+There are four options to choose from when it comes to the highlighter type:
+
+Default::
+The default highlighter type will use one of the three other highlighters described below.
+Which type will be used is backend-specific and might change in the future.
+For now, both Lucene and Elasticsearch backends will use the unified highlighter.
++
+Default highlighter exposes a set of simple configuration options available for all highlighter types.
+Plain::
+Plain highlighter can be useful for simple queries targeting a single field on a small number of documents.
+This highlighter uses a standard Lucene highlighter. It reads the string value of a highlighted field,
+then creates a small in-memory index from it and applies query logic to perform highlighting.
+Unified::
+Unified highlighter is used by default and does not necessarily rely on re-analyzing the text, as it can
+get the offsets either from postings or from term vectors.
++
+This highlighter uses a break iterator (breaks text into sentences by default) to break the text into later scored passages.
+It better supports more complex queries. Since it can work with prebuilt data, it performs better in case of a larger amount
+of documents compared to plain highlighter.
+Fast vector::
+Fast vector highlighter requires term vectors to be set to either `TermVector.WITH_POSITIONS_OFFSETS` or `TermVector.WITH_POSITIONS_OFFSETS_PAYLOADS`.
++
+In addition to using a break iterator similar to the unified highlighter, it can use the boundary characters
+to control the highlighted snippet.
++
+This is the only highlighter that can assign different weights to highlighted fragments, allowing it to show
+a fragment score differences by wrapping it with a different tag.
+For more on tags, see <<search-dsl-highlighting-highlighter-options-tags,the corresponding section>>.
+
+[NOTE]
+====
+If a full-text field that requires the `Highlightable.FAST_VECTOR` highlighter does not specify the term vectors,
+or set them to anything else besides the two mentioned options - then the term vectors for that field
+will get "upgraded" to one of `TermVector.WITH_POSITIONS_OFFSETS`/`TermVector.WITH_POSITIONS_OFFSETS_PAYLOADS`
+depending on the initially specified value.
+====
+
+Picking the highlighter type is the first step in highlighter definition:
+
+.Specifying highlighter type
+====
+[source, JAVA, indent=0, subs="+callouts"]
+----
+include::{sourcedir}/org/hibernate/search/documentation/search/highlighting/HighlightingDslIT.java[tags=various-highlighter-types]
+----
+<1> Starting a definition of a default highlighter.
+Configuration options are limited to the most common ones that can be found in other highlighters.
+<2> Starting a definition of a plain highlighter.
+<3> Starting a definition of a unified highlighter.
+<4> Starting a definition of a fast vector highlighter.
+====
+
+[[search-dsl-highlighting-highlighter-options]]
+== Highlighter options
```

Comment:
Ok. I wanted to have a common section for the options so that we could point to it like - "there are various config options see the section and pick whatever you'd like" ...
I guess we can keep the anchor and drop the section. it'll then take the user to the first option